src/build/emitter.ts

```diff
@@ -906,7 +907,10 @@ export function emitWebIdl(
       comments.push("Available only in secure contexts.");
     }
     if (entity.mdnUrl) {
-      if (comments.length > 0) comments.push("");
+      if (comments.length == 0) {
+        comments.push(extractSummaryFromFile(entity.mdnUrl));
```

Comment:
Emitter should just emit the typescript lib based on the given data, rather than generating data. Any required generation should be done in build.ts and passed to this function.